src/plugins/shared_dictionaries_plugin/shared_dictionaries_column_processor.cpp

```diff
@@ -0,0 +1,291 @@
+#include "shared_dictionaries_column_processor.hpp"
+
+namespace opossum {
+
+template <typename T>
+SharedDictionariesColumnProcessor<T>::SharedDictionariesColumnProcessor(
+    const std::shared_ptr<Table>& init_table, const std::string& init_table_name, const ColumnID init_column_id,
+    const std::string& init_column_name, const double init_jaccard_index_threshold,
+    const std::shared_ptr<SharedDictionariesPlugin::SharedDictionariesStats>& init_stats)
+    : table(init_table),
+      table_name(init_table_name),
+      column_id(init_column_id),
+      column_name(init_column_name),
+      jaccard_index_threshold(init_jaccard_index_threshold),
+      stats(init_stats) {}
+
+template <typename T>
+void SharedDictionariesColumnProcessor<T>::process() {
+  const auto allocator = PolymorphicAllocator<T>{};
+  auto merge_plans = std::vector<std::shared_ptr<MergePlan>>{};
+  _initialize_merge_plans(merge_plans);
+
+  std::optional<SegmentChunkPair<T>> previous_segment_chunk_pair_opt = std::nullopt;
+
+  const auto chunk_count = table->chunk_count();
+  for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
+    const auto chunk = table->get_chunk(chunk_id);
+    const auto segment = chunk->get_segment(column_id);
+    stats->total_previous_bytes += segment->memory_usage(MemoryUsageCalculationMode::Full);
```

Comment:
Have you checked the difference in terms of accuracy and runtime when changing the mode from `Full` to `Sampled`?

Reply:
In our tests we had an assertion error because of this check, if the mode was set to `Sampled`:
```
Assert(new_dictionary_memory_usage < previous_dictionary_memory_usage,
             "New dictionary memory usage should be lower than previous");
```

That's why we switched to use `Full`